<a href="https://colab.research.google.com/github/josh777-ops/Fly-Rank-AI/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/josh777-ops/Fly-Rank-AI/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I have chosen Lane 2 – Refresh / Content Opportunity Scoring.

I chose this lane because it addresses a practical business problem: deciding which content pages should be reviewed first when time and resources are limited. Instead of predicting something for its own sake, the project supports a real decision by ranking pages according to their likelihood of benefiting from review.

This lane also fits the starter dataset well because it already contains observable search and engagement signals such as impressions, clicks, sessions, CTR, average position, content age, and trend direction. These provide useful evidence for prioritising review candidates without relying on FlyRank's internal product decisions.

In [1]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Research Question

Which content pages should be reviewed first for refresh or improvement based on observable search and engagement signals?

Decision

Determine which pages should receive review priority.

Action

A content manager reviews the highest-ranked pages and decides whether to:

Refresh outdated content
Expand thin content
Improve titles and metadata
Improve user engagement
Monitor instead of editing
Unit of Analysis

One content page (one row represents one page).

Output

A ranked list of pages containing:

Refresh score
Suggested action
Reason codes
Confidence level
Cost of a Wrong Recommendation

If the model ranks the wrong pages too highly:

valuable editor time is wasted;
important declining pages may be missed;
opportunities for improving traffic could be delayed.

Because of this, the project is intended to support human decisions, not replace them.

Why Machine Learning Helps

There are many measurable signals that influence whether a page deserves attention. Machine learning can combine these signals more effectively than a small set of fixed rules and produce a better prioritised review queue.

In [2]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [4]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

df = pd.read_csv("content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Columns:", len(df.columns))

print(df["trend_direction"].value_counts())

print("Median impressions:", df["impressions_90d"].median())
print("Median sessions:", df["sessions_90d"].median())


Rows: 30000
Columns: 44
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64
Median impressions: 731.0
Median sessions: 7.0


The starter dataset contains 30,000 content pages with 44 variables describing search performance, engagement, content characteristics, and trend information.

The distribution of the target variable shows:

Down: 16,262 pages (54.2%)
Stable: 5,962 pages (19.9%)
Up: 4,388 pages (14.6%)
New: 2,236 pages (7.5%)
Flat: 1,152 pages (3.8%)

The dataset also shows that the median page receives 731 impressions and 7 sessions over the last 90 days.

These statistics suggest that the dataset contains a meaningful number of declining pages along with a wide range of visibility and engagement levels. This makes it suitable for building a system that ranks pages according to their priority for review rather than simply identifying whether a page is declining.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This project is a decision-support system, not proof that refreshing a page will increase traffic.

The model ranks pages according to observable evidence available in the dataset. It cannot prove that editing a page causes future improvement because this dataset is observational rather than experimental.

Any recommendations produced by the model should be reviewed by a human before action is taken.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.